var/spack/repos/builtin/packages/libceed/package.py

```diff
@@ -25,24 +25,23 @@ class Libceed(Package, CudaPackage, ROCmPackage):
 
     variant('occa', default=False, description='Enable OCCA backends')
     variant('debug', default=False, description='Enable debug build')
-    variant('libxsmm', default=False, description='Enable LIBXSMM backend')
-    variant('magma', default=False, description='Enable MAGMA backend')
+    variant('libxsmm', default=False, description='Enable LIBXSMM backend', when='@0.3:')
+    variant('magma', default=False, description='Enable MAGMA backend', when='@0.6:')
 
-    conflicts('+libxsmm', when='@:0.2')
-    conflicts('+magma', when='@:0.5')
     conflicts('+rocm', when='@:0.6')
 
-    depends_on('cuda', when='+cuda')
-    depends_on('hip@3.8.0', when='@0.7:0.7.99+rocm')
-    depends_on('hip@3.8.0:', when='@0.8:+rocm')
-    depends_on('hipblas@3.8.0:', when='@0.8:+rocm')
+    with when('+rocm'):
+        depends_on('hip@3.8.0', when='@0.7:0.7.99')
+        depends_on('hip@3.8.0:', when='@0.8:')
+        depends_on('hipblas@3.8.0:', when='@0.8:')
 
-    depends_on('occa@develop', when='@develop+occa')
-    depends_on('occa@1.1.0', when='@0.7:+occa')
-    depends_on('occa@1.0.8:', when='@0.4+occa')
-    depends_on('occa@1.0.0-alpha.5,develop', when='@:0.2+occa')
-    depends_on('occa+cuda', when='+occa+cuda')
-    depends_on('occa~cuda', when='+occa~cuda')
+    with when('+occa'):
+        depends_on('occa@develop', when='@develop')
```

Comment:
Perhaps it would be easier to combine the two PRs? Technically the OCCA backend no longer works, as of libCEED v0.9.0, so libceed@develop + occa@develop is broken and should not be available to users

Reply:
@jeremylt Let me know what you'd prefer. I can remove that line, or leave it and review one of the PRs you linked that does that. My point here was really getting rid of the `phases` attribute (I'm trying to check how many of the ~80 packages that declare custom phases really need them).

Reply:
@jedbrown, how about we merge this and I'll update my MR accordingly?

Reply:
Sounds good.